Feat/3944  add gocryptfs feature in docker (#3964)
tracim/tracim: tools_docker/README.md

⚠ After execute one of these command, Tracim will be reachable on your system on port 8080.

#### Running with gocryptfs encryption

Exemple with basic instance of Tracim (local usage with webdav and caldav) with encrypted storage:

for this exemple,
you first need to write password you want in `~/tracim/secret/password.txt` file.

```bash
echo "password" -> ~/tracim/secret/password.txt
```

```bash
docker run \

>        -it \

Reviewer: is `-it` wanted here?

Reviewer 2: Probably not, need to be tested with and without to verify

>        -it \

Reviewer 2: done .

       --device /dev/fuse
       --cap-add SYS_ADMIN
       --security-opt apparmor:unconfined
       -e DATABASE_TYPE=sqlite \
       -e ENABLE_GOCRYPTFS_ENCRYPTION=1 \
       -e GOCRYPTFS_PREVIEW_STORAGE_DIR=/var/tracim/previews \
       -e TRACIM_PREVIEW_CACHE_DIR=/media/previews \
       -e GOCRYPTFS_UPLOADED_FILES_STORAGE_DIR=/var/tracim/uploaded_files \
       -e TRACIM_DEPOT_STORAGE_DIR=/media/uploaded_files \
       -e GOCRYPTFS_PASSWORD_PATH=/var/secret/password.txt \
       -p 8081:80 \
       -v ~/tracim/etc:/etc/tracim \
       -v ~/tracim/var:/var/tracim \
       -v ~/tracim/secret:/var/secret \
       algoo/tracim
```

> if you want a bit more security, you can remove password file.

Reviewer: ```suggestion
if you want a bit more security, you may want to remove the password file.
```

You will need it each time you need to run the docker.

```bash
rm ~/tracim/secret/password.txt
```

### Build images

To build image

    cd tools_docker/Debian_Uwsgi
    docker build -t algoo/tracim:latest .

> To build encrypted-enabled image (beta):

Reviewer: ```suggestion
To build encryption-enabled image (beta):
```

⚠ After execute one of these command, Tracim will be reachable on your system on port 8080.

#### Running with gocryptfs encryption

This need the new specific Debian_New_Uwsgi docker (see build section). This is an experimental
docker image, the new feature from this docker will maybe be merged to the standard docker or removed.
Exemple with basic instance of Tracim (local usage with webdav and caldav) with encrypted storage:

for this exemple,
you first need to write password you want in `~/tracim/secret/password.txt` file.

```bash

> echo "password" -> ~/tracim/secret/password.txt

Reviewer: ```suggestion
echo "password" > ~/tracim/secret/password.txt
```

Warn that it writes the password in the bash history and that people should edit the password with a text editor instead

⚠ After execute one of these command, Tracim will be reachable on your system on port 8080.

#### Running with gocryptfs encryption

This need the new specific Debian_New_Uwsgi docker (see build section). This is an experimental
docker image, the new feature from this docker will maybe be merged to the standard docker or removed.
Exemple with basic instance of Tracim (local usage with webdav and caldav) with encrypted storage:

for this exemple,
you first need to write password you want in `~/tracim/secret/password.txt` file.

```bash

> echo "password" > ~/tracim/secret/password.txt

Reviewer: ```suggestion
mkdir -p ~/tracim/secret
echo "password" > ~/tracim/secret/password.txt
```

> ```

Reviewer 3: Where is this path when you start Tracim with docker ?

Reviewer 3: This path is created after starting docker (in next point). I not understand this step.

Reviewer 2: You need a password file first to make this work, so the mount point and password.txt should exist.

Note: this is just an example, we suggest you to write password with a text editor
instead in order to not store the password in the bash history.

```bash
docker run \
       --device /dev/fuse
       --cap-add SYS_ADMIN
       --security-opt apparmor:unconfined
       -e DATABASE_TYPE=sqlite \
       -e ENABLE_GOCRYPTFS_ENCRYPTION=1 \

>        -e GOCRYPTFS_PREVIEW_STORAGE_DIR=/var/tracim/previews \

Reviewer 3: Why this path is not the same like we use by default ? `/var/tracim/data/preview`

Reviewer 2: you're right.

>        -e TRACIM_PREVIEW_CACHE_DIR=/media/previews \

Reviewer 3: Why this is not in volume ?

Reviewer 2: This is the unencrypted mount point, there is no need to make this in volume (it's also less secure)

>        -e GOCRYPTFS_UPLOADED_FILES_STORAGE_DIR=/var/tracim/uploaded_files \

Reviewer 3: why `uploaded_files` is not in data folder ? `uploaded_files` is not for data ?

Reviewer 3: same like preview, why not in /var/tracim/data path ?

Reviewer 2: you're right. should we do this now or wait for refactor ?

Reviewer 2: done.

       -e TRACIM_DEPOT_STORAGE_DIR=/media/uploaded_files \
       -e GOCRYPTFS_PASSWORD_PATH=/var/secret/password.txt \

>        -p 8081:80 \

Reviewer 3: Why not use the standard port we have define for tracim. default value `-p 8080:80`.

Reviewer 2: right.

Reviewer 2: done.
